Can't reshape with shape of another pyarray

```c++
inline auto add(xt::pyarray<double> &a, xt::pyarray<double> &b)
{
    auto res = xt::pyarray<double>();
    res.reshape(a.shape());
    xt::noalias(res) = (a + b);
    return res;
}
```

fails to compile with:

```
src/main.cpp: In function ‘auto add(xt::pyarray<double>&, xt::pyarray<double>&)’:
src/main.cpp:26:26: error: no matching function for call to ‘xt::pyarray<double>::reshape(const inner_shape_type&)’
     res.reshape(a.shape());
                          ^
In file included from /home/wolfv/Programs/xtensor-python/include/xtensor-python/pyarray.hpp:20:0,
                 from src/main.cpp:8:
/home/wolfv/Programs/xtensor-python/include/xtensor-python/pycontainer.hpp:222:17: note: candidate: void xt::pycontainer<D>::reshape(const shape_type&) [with D = xt::pyarray<double>; xt::pycontainer<D>::shape_type = std::vector<long unsigned int>]
     inline void pycontainer<D>::reshape(const shape_type& shape)
                 ^~~~~~~~~~~~~~
/home/wolfv/Programs/xtensor-python/include/xtensor-python/pycontainer.hpp:222:17: note:   no known conversion for argument 1 from ‘const inner_shape_type {aka const xt::xbuffer_adaptor<long unsigned int>}’ to ‘const shape_type& {aka const std::vector<long unsigned int>&}’
/home/wolfv/Programs/xtensor-python/include/xtensor-python/pycontainer.hpp:236:17: note: candidate: void xt::pycontainer<D>::reshape(const shape_type&, xt::layout_type) [with D = xt::pyarray<double>; xt::pycontainer<D>::shape_type = std::vector<long unsigned int>]
     inline void pycontainer<D>::reshape(const shape_type& shape, layout_type l)
                 ^~~~~~~~~~~~~~
/home/wolfv/Programs/xtensor-python/include/xtensor-python/pycontainer.hpp:236:17: note:   candidate expects 2 arguments, 1 provided
/home/wolfv/Programs/xtensor-python/include/xtensor-python/pycontainer.hpp:249:17: note: candidate: void xt::pycontainer<D>::reshape(const shape_type&, const strides_type&) [with D = xt::pyarray<double>; xt::pycontainer<D>::shape_type = std::vector<long unsigned int>; xt::pycontainer<D>::strides_type = std::vector<long unsigned int>]
     inline void pycontainer<D>::reshape(const shape_type& shape, const strides_type& strides)
                 ^~~~~~~~~~~~~~
/home/wolfv/Programs/xtensor-python/include/xtensor-python/pycontainer.hpp:249:17: note:   candidate expects 2 arguments, 1 provided
```

--- a/include/xtensor-python/pycontainer.hpp
+++ b/include/xtensor-python/pycontainer.hpp
@@ -225,7 +225,7 @@ namespace xt
     template <class S>
     inline void pycontainer<D>::reshape(const S& shape)
     {
-        if (shape.size() != this->dimension() || !std::equal(shape.begin(), shape.end(), this->shape().begin()))
+        if (shape.size() != this->dimension() || !std::equal(std::begin(shape), std::end(shape), std::begin(this->shape())))
         {
             reshape(shape, layout_type::row_major);
         }
@@ -254,7 +254,7 @@ namespace xt
     template <class S>
     inline void pycontainer<D>::reshape(const S& shape, const strides_type& strides)
     {
-        derived_type tmp(shape, strides);
+        derived_type tmp(xtl::forward_sequence<shape_type>(shape), strides);
         *static_cast<derived_type*>(this) = std::move(tmp);
     }
 

--- a/test/test_common.hpp
+++ b/test/test_common.hpp
@@ -130,7 +130,7 @@ namespace xt
             m_data = {-1, 1, 2, 3, 8, 9, 10, 11, 16, 17, 18, 19};
             m_layout = layout_type::dynamic;
             m_assigner.resize(m_shape[0]);
-            for (std::size_t i = 0; i < m_shape[0]; ++i)
+            for (std::size_t i = 0; i < std::size_t(m_shape[0]); ++i)
             {
                 m_assigner[i].resize(m_shape[1]);
             }
@@ -184,10 +184,10 @@ namespace xt
             auto v_copy_b = vec;
             std::array<std::size_t, 3> ar = {3, 2, 4};
             std::vector<std::size_t> vr = {3, 2, 4};
-            // v_copy_a.reshape(ar, true);
-            // compare_shape(v_copy_a, rm);
-            // v_copy_b.reshape(vr, true);
-            // compare_shape(v_copy_b, rm);
+            v_copy_a.reshape(ar);
+            compare_shape(v_copy_a, rm);
+            v_copy_b.reshape(vr);
+            compare_shape(v_copy_b, rm);
         }
 
         {

--- a/test/test_pyarray.cpp
+++ b/test/test_pyarray.cpp
@@ -161,6 +161,10 @@ namespace xt
     {
         pyarray<int> a;
         test_reshape(a);
+
+        pyarray<int> b = { {1, 2}, {3, 4} };
+        a.reshape(b.shape());
+        EXPECT_EQ(a.shape(), b.shape());
     }
 
     TEST(pyarray, transpose)

--- a/test/test_pytensor.cpp
+++ b/test/test_pytensor.cpp
@@ -160,6 +160,10 @@ namespace xt
     {
         pytensor<int, 3> a;
         test_reshape<pytensor<int, 3>, container_type>(a);
+
+        pytensor<int, 3> b = { { { 1, 2 },{ 3, 4 } } };
+        a.reshape(b.shape());
+        EXPECT_EQ(a.shape(), b.shape());
     }
 
     TEST(pytensor, transpose)

